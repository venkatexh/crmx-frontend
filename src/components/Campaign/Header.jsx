import "../../sass/components/contact/contact.component.scss";

const Header = ({campaign}) => {
  const {name, from, status} = campaign;
  return <div className={'contactHeader'}>
    <div className={'left'}>
      <div className={'name'}>{name}</div>
      <div>From {from}</div>
    </div>
    <div>
      <div>Added 2nd Jan 2022</div>
      <div>Last Updated 2nd Jan 2022</div>
    </div>
    <div>Actions</div>
    <div>{status}</div>
  </div>
}

export default Header;