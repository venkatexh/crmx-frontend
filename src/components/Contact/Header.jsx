import "../../sass/components/contact/contact.component.scss";

const Header = ({contact}) => {
  const {firstName, lastName, email, status} = contact;
  return <div className={'contactHeader'}>
    <div className={'left'}>
      <div className={'name'}>{firstName} {lastName}</div>
      <div>{email}</div>
    </div>
    <div>
      <div>Added 2nd Jan 2022</div>
      <div>Last Updated 2nd Jan 2022</div>
    </div>
    <div>Actions</div>
    <div>{status}</div>
  </div>
}

export default Header;