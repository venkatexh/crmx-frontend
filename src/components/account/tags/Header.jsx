import { useDispatch, useSelector } from "react-redux";
import "../../../sass/components/account/dashboard/header.component.scss";
import { Link } from "react-router-dom";
import { toggleModal } from "../../../redux/actions/utility/toggleModal";

const Header = () => {
  const state = useSelector(({ loggedUser }) => ({ loggedUser }));
  const dispatch = useDispatch();
  const modalType = "new-tag";
  const handleClick = () => {
    dispatch(toggleModal(modalType));
  };
  return (
    <div className={"dashHeader"}>
      <div>
        <div className={"greeting"}>Contacts</div>
        <div className={"contactCount"}>
          You have <span className={"highlight"}>1414</span> tags in{" "}
          {state.loggedUser.company}
        </div>
      </div>
      <div>
        <Link to={"contacts"}>
          <button className={"viewContactsBtn"} onClick={() => handleClick()}>
            New Tag
          </button>
        </Link>
      </div>
    </div>
  );
};

export default Header;
