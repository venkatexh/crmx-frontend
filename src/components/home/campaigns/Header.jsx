import { useDispatch } from "react-redux";
import "../../../sass/components/home/dashboard/header.component.scss";
import { toggleModal } from "../../../redux/actions/utility/toggleModal";

const Header = () => {
  // const state = useSelector(({ loggedUser }) => ({ loggedUser }));
  const dispatch = useDispatch();
  const modalType = "new-campaign";
  const handleClick = () => {
    dispatch(toggleModal(modalType));
  };
  return (
    <div className={"dashHeader"}>
      <div>
        <div className={"greeting"}>Campaigns</div>
        <div className={"contactCount"}>
          You have <span className={"highlight"}>1414</span> campaigns
        </div>
      </div>
      <div>
        {/*<Link to={"contacts"}>*/}
        <button className={"viewContactsBtn"} onClick={() => handleClick()}>
          New Campaign
        </button>
        {/*</Link>*/}
      </div>
    </div>
  );
};

export default Header;
