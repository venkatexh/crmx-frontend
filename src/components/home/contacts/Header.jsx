import {useDispatch, useSelector} from "react-redux";
import "../../../sass/components/home/dashboard/header.component.scss";
import {Link} from "react-router-dom";
import {toggleModal} from "../../../redux/actions/utility/toggleModal";

const Header = () => {
  const state = useSelector(({loggedUser}) => ({loggedUser}));
  console.log(state.loggedUser)
  const dispatch = useDispatch();
  const modalType = "new-contact";
  const handleClick = () => {
    dispatch(toggleModal(modalType));
  };
  return (
    <div className={"dashHeader"}>
      <div>
        <div className={"greeting"}>Contacts</div>
        <div className={"contactCount"}>
          You have <span className={"highlight"}>1414</span> contacts in{" "}
          {state.loggedUser.company}
        </div>
      </div>
      <div>
        <Link to={"contacts"}>
          <button className={"viewContactsBtn"} onClick={() => handleClick()}>
            New Contact
          </button>
        </Link>
      </div>
    </div>
  );
};

export default Header;
