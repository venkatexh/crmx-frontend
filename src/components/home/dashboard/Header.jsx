import {useDispatch, useSelector} from "react-redux";
import "../../../sass/components/home/dashboard/header.component.scss";
import {Link} from "react-router-dom";
import {changeWindow} from "../../../redux/actions/account/changeWindow";

const Header = () => {
  const state = useSelector(({loggedUser}) => ({loggedUser}));
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(changeWindow("contacts"));
  };
  return (
    <div className={"dashHeader"}>
      <div>
        <div className={"greeting"}>Hi {state.loggedUser.firstName}!</div>
        <div className={"contactCount"}>
          You have <span className={"highlight"}>1414</span> contacts in{" "}
          {state.loggedUser.company}
        </div>
      </div>
      <div>
        <Link to={"contacts"}>
          <button className={"viewContactsBtn"} onClick={() => handleClick()}>
            View Contacts
          </button>
        </Link>
      </div>
    </div>
  );
};

export default Header;
