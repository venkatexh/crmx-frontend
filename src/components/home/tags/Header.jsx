import { useDispatch, useSelector } from "react-redux";
import "../../../sass/components/home/dashboard/header.component.scss";
import { toggleModal } from "../../../redux/actions/utility/toggleModal";

const Header = () => {
  const state = useSelector(({ loggedUser }) => ({ loggedUser }));
  const dispatch = useDispatch();
  const modalType = "new-tag";
  const handleClick = () => {
    dispatch(toggleModal(modalType));
  };
  return (
    <div className={"dashHeader"}>
      <div>
        <div className={"greeting"}>Contacts</div>
        <div className={"contactCount"}>
          You have <span className={"highlight"}>1414</span> tags in{" "}
          {state.loggedUser.company}
        </div>
      </div>
      <div>
        <button className={"viewContactsBtn"} onClick={() => handleClick()}>
          New Tag
        </button>
      </div>
    </div>
  );
};

export default Header;
