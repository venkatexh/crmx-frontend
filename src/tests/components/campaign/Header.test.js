import { shallow } from "enzyme";
import React from "react";
import Header from "../../../components/Campaign/Header";

it("Campaign header component snapshot", () => {
  expect(
    shallow(
      <Header
        campaign={{ name: "Test campaign", status: "sent", from: "John Doe" }}
      />
    )
  ).toMatchSnapshot();
});
