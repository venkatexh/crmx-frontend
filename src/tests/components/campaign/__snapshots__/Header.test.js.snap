// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign header component snapshot 1`] = `ShallowWrapper {}`;
